tests/indices/query/test_compose_vector.py

```diff
@@ -1,31 +1,30 @@
 """Test recursive queries."""
 
+import pytest
 import asyncio
 from pathlib import Path
+import sys
 from tempfile import TemporaryDirectory
 from typing import Any, Dict, List, Tuple, cast
-from unittest.mock import patch
-
-import pytest
+from unittest.mock import MagicMock, patch
 
 from gpt_index.data_structs.data_structs_v2 import V2IndexStruct
 from gpt_index.data_structs.struct_type import IndexStructType
 from gpt_index.embeddings.openai import OpenAIEmbedding
 from gpt_index.indices.composability.graph import ComposableGraph
 from gpt_index.indices.keyword_table.simple_base import GPTSimpleKeywordTableIndex
-from gpt_index.indices.list.base import GPTListIndex
 from gpt_index.indices.query.schema import QueryConfig, QueryMode
-from gpt_index.indices.tree.base import GPTTreeIndex
-from gpt_index.indices.vector_store.vector_indices import GPTSimpleVectorIndex
+from gpt_index.indices.vector_store.vector_indices import (
+    GPTPineconeIndex,
+    GPTSimpleVectorIndex,
+)
 from gpt_index.langchain_helpers.chain_wrapper import (
-    LLMChain,
-    LLMMetadata,
     LLMPredictor,
 )
 from gpt_index.langchain_helpers.text_splitter import TokenTextSplitter
 from gpt_index.readers.schema.base import Document
+from tests.indices.vector_store.test_pinecone import MockPineconeIndex
```

Comment:
nit: would probably abstract this class out into like a tests.indices.vector_store.utils file but not a huge deal rn

Reply:
moved